extensions/replication/queueProcessor/MultipleBackendTask.js

```diff
@@ -380,112 +380,48 @@ class MultipleBackendTask extends QueueProcessorTask {
     }
 
-    _getAndPutObjectTaggingOnce(sourceEntry, destEntry, log, cb) {
+    _putObjectTaggingOnce(sourceEntry, destEntry, log, cb) {
         const doneOnce = jsutil.once(cb);
         log.debug('replicating object tags', {
             entry: sourceEntry.getLogInfo(),
         });
-        const sourceReq = this.S3source.getObjectTagging({
-            Bucket: sourceEntry.getBucket(),
-            Key: sourceEntry.getObjectKey(),
-            VersionId: sourceEntry.getEncodedVersionId(),
-        });
-        attachReqUids(sourceReq, log);
-        sourceReq.on('error', err => {
-            // eslint-disable-next-line no-param-reassign
-            err.origin = 'source';
-            if (err.statusCode === 404) {
-                log.error('the source object was not found', {
-                    method: 'MultipleBackendTask._getAndPutObjectTaggingOnce',
-                    entry: sourceEntry.getLogInfo(),
-                    origin: 'source',
-                    peer: this.sourceConfig.s3,
-                    error: err.message,
-                    httpStatus: err.statusCode,
-                });
-                return doneOnce(err);
-            }
-            log.error('an error occurred on getObject from S3', {
-                method: 'MultipleBackendTask._getAndPutObjectTaggingOnce',
-                entry: sourceEntry.getLogInfo(),
-                origin: 'source',
-                peer: this.sourceConfig.s3,
-                error: err.message,
-                httpStatus: err.statusCode,
+        const destReq = this.backbeatSource
+            .multipleBackendPutObjectTagging({
+                Bucket: destEntry.getBucket(),
+                Key: destEntry.getObjectKey(),
+                SourceBucket: sourceEntry.getBucket(),
+                SourceVersionId: sourceEntry.getVersionId(),
+                StorageType: destEntry.getReplicationStorageType(),
+                StorageClass: destEntry.getReplicationStorageClass(),
+                Tags: JSON.stringify(destEntry.getTags()),
             });
-            return doneOnce(err);
-        });
-        const incomingMsg = sourceReq.createReadStream();
-        incomingMsg.on('error', err => {
-            if (err.statusCode === 404) {
-                log.error('the source object was not found', {
-                    method: 'MultipleBackendTask._getAndPutObjectTaggingOnce',
-                    entry: sourceEntry.getLogInfo(),
-                    origin: 'source',
-                    peer: this.sourceConfig.s3,
+        attachReqUids(destReq, log);
+        return destReq.send((err, data) => {
+            if (err) {
+                // eslint-disable-next-line no-param-reassign
+                err.origin = 'source';
+                log.error('an error occurred putting object tagging to S3', {
+                    method: 'MultipleBackendTask._putObjectTaggingOnce',
+                    entry: destEntry.getLogInfo(),
+                    origin: 'target',
```

Comment:
is it intentional that the origin for the error and here differ?

Reply:
It is doing nothing here so I've removed it. 

The `err.origin` is set to 'source' throughout this task so that in a retry it never enters [this kind of condition](https://github.com/scality/backbeat/blob/master/extensions/replication/queueProcessor/QueueProcessorTask.js#L156) (as this class is inherited). For the log though, the error should come from AWS so I've left it at as 'target'. It looks like all these could removed as the check is `err.origin === 'target'`.